ckanext/unaids/plugin.py

```diff
@@ -238,6 +238,11 @@ def before_create(self, context, resource):
         return self._process_schema_fields(resource)
 
     def before_update(self, context, current, resource):
+        file_uploaded = current.get("url") != resource.get("url")
+
+        if file_uploaded and resource.get("validate_package"):
+            self.resources_to_validate_package[resource['id']] = True
+
```

Comment:
This block should be inside `if _data_dict_is_resource()` as in CKAN 2.9 the same `before_update` is still called for dataset & resource actions.

Reply:
Yes, whilst not needed for functionality, it will definitely improve readability and future proof it agains tht epossibility of someone adding a url field to the dataset schema. I like this.